cmd/edit/cluster/cmd.go

```diff
@@ -1142,3 +1189,163 @@ func confirmMigration() (bool, error) {
+	return cluster.AutoNode() != nil && cluster.AutoNode().Mode() == "enabled"
+}
+
+func autonodeRoleExists(cluster *cmv1.Cluster) string {
+	if cluster.AWS() != nil && cluster.AWS().AutoNode() != nil {
+		return cluster.AWS().AutoNode().RoleArn()
+	}
+	return ""
+}
+
+func setAutoNode(r *rosa.Runtime, cmd *cobra.Command, cluster *cmv1.Cluster, autonode string, roleArn string) (
+	mode string, role string, err error) {
+
+	// Check if flags are changed
+	autonodeChanged := cmd.Flags().Changed("autonode")
+	roleArnChanged := cmd.Flags().Changed("autonode-iam-role-arn")
+
+	if !autonodeChanged && !roleArnChanged {
+		return "", "", nil
+	}
+
+	// Validate HCP cluster
+	if !aws.IsHostedCP(cluster) {
+		return "", "", fmt.Errorf("AutoNode is only supported for Hosted Control Plane clusters")
+	}
+
+	// Get current state
+	currentAutoNodeEnabled := autonodeExists(cluster)
+	currentRoleArn := autonodeRoleExists(cluster)
+
+	// Validate autonode value
+	if autonodeChanged {
+		if autonode != "enabled" {
+			return "", "", fmt.Errorf("Invalid value for --autonode. Currently only 'enabled' is supported")
+		}
+		mode = autonode
+	}
+
+	// Handle role ARN
+	if roleArnChanged {
+		if roleArn != "" && !aws.RoleArnRE.MatchString(roleArn) {
+			return "", "", fmt.Errorf("Invalid IAM Role ARN format: %s", roleArn)
+		}
+		role = roleArn
+	}
+
+	// If enabling AutoNode, role ARN is required
+	if mode == "enabled" && !currentAutoNodeEnabled {
+		if role == "" && currentRoleArn == "" {
+			return "", "", fmt.Errorf("IAM Role ARN is required when enabling AutoNode. Use --autonode-iam-role-arn to specify the role")
+		}
+		if role == "" {
+			role = currentRoleArn
+		}
+	}
+
+	// If only updating role ARN, AutoNode must already be enabled
+	if roleArnChanged && !autonodeChanged && !currentAutoNodeEnabled {
+		return "", "", fmt.Errorf("Cannot update AutoNode IAM Role ARN when AutoNode is not enabled. Enable AutoNode first using --autonode=enabled")
+	}
+
+	// If updating role ARN with AutoNode already enabled, maintain enabled state
+	if roleArnChanged && currentAutoNodeEnabled && !autonodeChanged {
+		mode = "enabled"
+	}
+
+	// Confirm with user
+	if mode != "" || role != "" {
+		confirmAutoNode(r, mode, role, currentAutoNodeEnabled, currentRoleArn)
+	}
+
+	return mode, role, nil
+}
+
+func confirmAutoNode(r *rosa.Runtime, mode string, roleArn string, currentEnabled bool, currentRoleArn string) {
+	if mode == "enabled" && !currentEnabled {
+		r.Reporter.Warnf("You are choosing to enable AutoNode for automatic node management")
+		if !confirm.Confirm("enable AutoNode for cluster with the provided role arn '%s'", roleArn) {
+			os.Exit(0)
+		}
+	} else if roleArn != "" && currentRoleArn != "" && roleArn != currentRoleArn {
+		r.Reporter.Warnf("You are choosing to update the AutoNode IAM role ARN")
+		if !confirm.Confirm("update AutoNode IAM role ARN from '%s' to '%s'", currentRoleArn, roleArn) {
+			os.Exit(0)
+		}
+	}
```

Comment:
while encapsulating code is good for not duplicating it- I think we should move this code to where the function is called and delete the function.

I have a couple of reasons;

1) This function is only an if statement controlled by a boolean passed in,[ I think this article can explain it better than I can though (4 minute read)](https://alexkondov.com/should-you-pass-boolean-to-functions/)
2) This code is not going to be reusable really, we only need to call this confirmation once and it's very specific. But, if we were to reuse it, we should make 2 functions and put each branch into a separate function

WDYT?

Reply:
You're right, the article makes a good case, this is indeed a code smell, fixed on the last commit